### PR TITLE
feat: socket timeout

### PR DESCRIPTION
**Pull request by lucasantoiemma**
By the nodejs API documentation the Socket object doesn't have a timeout set.
I added a new param socketTimeout (numeric) to pass to the FtpServer so the timeout can be set on the socket created.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -39,7 +40,11 @@ class FtpServer extends EventEmitter {
       _.get(this, 'options.pasv_min'),
       _.get(this, 'options.pasv_max'));
 
+	const timeout = Number(this.options.socketTimeout);
+	this.socketTimeout = isNaN(timeout) ? 0 : timeout;
```
Comment (trs) [inline src/index.js]: No need to set a whole new variable when we have it already set under `options`.

File: src/index.js [hunk 2]
```diff
@@ -23,7 +23,8 @@ class FtpServer extends EventEmitter {
       blacklist: [],
       whitelist: [],
       greeting: null,
-      tls: false
+      tls: false,
+      socketTimeout: 0
```
Comment (trs) [inline src/index.js]: I think just `timeout` would be fine instead.

File: src/index.js (hunk 1 above)
Comment (lucasantoiemma) [reply src/index.js]: Someone prefers the split version to increase readability, but it makes sense. No problem

File: src/index.js [hunk 3]
```diff
@@ -39,7 +40,10 @@ class FtpServer extends EventEmitter {
       _.get(this, 'options.pasv_min'),
       _.get(this, 'options.pasv_max'));
 
+	this.timeout = isNaN(Number(this.options.timeout)) ? 0 : Number(this.options.timeout);
```
Comment (trs) [inline src/index.js]: I liked having a scoped variable to get the timeout. But set it on the options timeout, not a timeout directly on the instance.

```suggestion
      const timeout = Number(this.options.timeout);
      this.options.timeout = isNaN(timeout) ? 0 : Number(timeout);
```

File: src/index.js [hunk 4]
```diff
@@ -39,7 +40,10 @@ class FtpServer extends EventEmitter {
...
+
     const serverConnectionHandler = (socket) => {
+      socket.setTimeout(this.timeout);
```
Comment (trs) [inline src/index.js]:
```suggestion
      socket.setTimeout(this.options.timeout);
```

**Comment by trs** [comment]
Thank you for the PR, looks great!

**Pull request merged**